### DISCUSSION TITLE
drawTriangle and fillTriangle

### DISCUSSION THREAD

**Discussion by maho24at** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#534] [Q&A]
Hi guys! 

first of all!!! THANK YOU FOR THIS AWESOME LIBRARY!!!!!!!

**am I blind or is there no function for triangles?** 

how should I do? drawing lines is maybe not so efficient...

**Comment by board707** [comment]
> drawing lines is maybe not so efficient...

What method, besides drawing lines, can you draw a triangle?

**Comment by maho24at** [comment]
i was an idiot,.. 

when I tried the fillTriangle function first, I must have made an mistake because it did not work, then I searched the whole library for the function could not find it,.. then I posted here, and then I found out that that I can use the adafruit GFX library functions,...

for others,.. just use.. 

dma_display->fillTriangle(0, 0, 191, 0, 192, 63, dma_display->color444(255, 255, 255));


is there a way to stop the flickering when using triangles ? I draw a triangle for 0.5s

**Reply by mrcodetastic** [reply]
This library isn't a graphics library, buy has a dependency on (and uses the functions from) the Adafruit GFX library to "tell" this library what LEDs on the physical HUB75 LED Matrix need to be lit.
